### ISSUE TITLE
Problem: Project shall have one branch, `master`, and no topic branches

### ISSUE THREAD

**Issue by LEDCoyote** [magicstone-dev/ecko#147]
### Problem
When analyzing the structure of this repository, I note that the repository is running on the `c4` branch as its default, instead of `main` or `master` and that the fork contains numerous topic branches. This is fine and well for a personal fork, but for C4 it isn't compliant, due to
> 2.5. Branches and Releases
>   1. The project SHALL have one branch ("master") that always holds the latest in-progress version and SHOULD always build.
>   2. The project SHALL NOT use topic branches for any reason. Personal forks MAY use topic branches.

### Proposed Solution
Create a Github organization (called "C4Social", perhaps), which @weex administers, and which holds the clean project repository, allowing https://github.com/weex/mastodon/ to be a personal fork of it. _This repository_ could be [transferred](https://docs.github.com/en/github/administering-a-repository/managing-repository-settings/transferring-a-repository) to the organization to be the main project repo, and would retain its issues, PRs, etc. After that, clean up the branches of `{org-name}/mastodon`.

_Update (2021-08-21):_ rewrote this issue in "problem/solution" format.

**Comment by LEDCoyote** [comment]
Thinking about this further, it seems like having the `main` branch along with `c4` could be seen as helping with interacting with upstream, mainline Mastodon. Wherein `main` could be kept up-to-date with mastodon/mastodon. And changes in that branch could be pulled into `c4`, where we enforce the C4 process magic.

Not sure that it's actually of value, though, since in that case our `main` is simply redundant with mainline's `main` branch. The workflow I would imagine for pulling in changes from mainline would be something like:
1. Merge mainline main into a personal development fork.
2. Test and ensure operability/desirability of changes
3. Make changes and iterate until correct, if necessary
4. PR into the organization project repo

The above would have to be done by Contributors at a cadence close to lockstep with mainline's velocity, to ensure we don't diverge too far from mainline and that the PR in step 4 doesn't get too large. Those are just some thoughts on process and structure, and I'm open to feedback and criticism of them (note: I'm pretty new to open source development, so feel free to enlighten me if there are better established patterns for this).

**Comment by weex** [comment]
Great start! I agree, though there are a couple of different problems described here. Sticking to the title as "the problem" for this thread, I can say the c4 branch was an idiosyncrasy of this initial bring up and should become `main` in a C4Social org-owned repo. Also agree with the elimination of the topic branches, again an artifact of an informal start.

C4's reference to `master` is obsolete with `main` being the more inclusive default branch name. I'm not sure if that's cause to fork the C4 rfc itself but for now, I'll create an addendum stating that `main` is the new `master`.

The other thing mentioned above is the syncing process with upstream and for that it'd be best to make an issue in https://github.com/weex/C4-tools, which should also be rehomed in the org.

**Comment by weex** [comment]
Repos have been transferred.

The topic branches that exist are for open UpstreamPRs but those PRs can be closed and topic branches deleted. They still exist in my personal fork and I'll need to recreate them. I made comments on upstream PRs in the case that there were merge conflicts so this will be a good opportunity to revisit them and see if any have become mergeable.

**Comment by LEDCoyote** [comment]
+1 project in c4social organization repo
+1 `main` > `master` Fantastic, in total agreement on the inclusivity aspect here.

> there are a couple of different problems described here. 

Yup, I got a little meandering in my thoughts here. I'll think about whether there's anything in the remaining parts that might fit into a separate issue, or whether it will take care of itself. I'm sorta expecting the latter to be the case. 

I'll close this issue once `main` is up-to-date and the extraneous branches get pruned. I ought to have a little time to look at that this evening 👍

**Comment by LEDCoyote** [comment]
>The other thing mentioned above is the syncing process with upstream and for that it'd be best to make an issue in https://github.com/weex/C4-tools, which should also be rehomed in the org.

I'll dig into that codebase and think about how to formulate the issue w.r.t. those tools.

**Comment by weex** [comment]
I've deleted all topic branches that were straightforward to delete.

The current list of branches includes:
`c4` - our current default branch, which can be renamed to `main` once a PR is merged on c4-tools to target main
`vagrant-import-from-rvmio` - has open pr with upstream, can delete once merged or closed
`docker-no-root-warn` - has open pr with upstream, can delete once merged or closed

Many of the deleted topic branches corresponded to PRs from before the repo was transferred. They still exist on my personal fork and I've opened PRs from there. Also adding notes to those individual PRs so it easier to deal with each one individually.

**Comment by LEDCoyote** [comment]
> so c4 will be renamed with @LEDCoyote's ack.

*Picard voice*: Make it so.

Great work clearing branches!

**Comment by weex** [comment]
Down to just one branch `ignore-matching-hashtags` which is waiting for https://github.com/mastodon/mastodon/pull/16460

**Comment by weex** [comment]
Yay that was merged in Mastodon so now we have:

<img width="222" alt="Screen Shot 2021-12-15 at 3 29 35 PM" src="https://user-images.githubusercontent.com/38776/146280820-25b1e777-6192-4c84-b11a-ed5ed697087e.png">
(image of Github UI showing just 1 branch)
